pinot-controller/src/main/java/org/apache/pinot/controller/validation/RealtimeSegmentValidationManager.java

```diff
@@ -169,6 +171,10 @@ private void runSegmentLevelValidation(TableConfig tableConfig, StreamConfig str
     if (_llcRealtimeSegmentManager.isDeepStoreLLCSegmentUploadRetryEnabled()) {
       _llcRealtimeSegmentManager.uploadToDeepStoreIfMissing(tableConfig, segmentsZKMetadata);
     }
+
+    if (_segmentAutoResetOnErrorAtValidation) {
+      _pinotHelixResourceManager.resetSegments(realtimeTableName, null, true);
+    }
```

Comment:
This applies to all segments - consuming and otherwise right? What kinds of scenarios would this auto reset mechanism help with?

Reply:
This would only apply to segments that are in an error state. The definition of the `resetSegments` method:
```
public void resetSegments(String tableNameWithType, @Nullable String targetInstance, boolean errorSegmentsOnly) {
```

Reply:
Adding a bit more background, for some of our largest clusters (>1M segments per zk) we sporadically find segments in error state. Many of these can be fixed with a simple reset and we would like to avoid operator intervention for these cases.

One example case: 
1. server 1 completes build, fails to upload to deep store
2. server 2 is being restarted/upgraded/replaced, when it starts up peer download fails
3. server 1 backfills deepstore via async upload task
4. server 2's segment needs to be reset to trigger deep-store download and load the segment 

We have tried increasing deep store upload/peer download timeouts/retries, but this isn't a great solution for us since it introduces more delays into the ingestion path

Reply:
@shauryachats I understand that, I was curious about the specific error cases you think this mechanism would be useful for.

@itschrispeck thanks for the detailed background!

Reply:
This can be applied to both OFFLINE and REALTIME tables. Also, ideally we should trigger it only when we find ERROR segments in EV. Every call will fetch IS and EV from ZK